modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -257,6 +265,34 @@ public void onConnectionException(ConnectionException e) {
       return watermarkComparator;
     }
 
+    private void setUpdatedWatermark(Serializable updatedWatermark) {
+      try {
+        this.updatedWatermark = updatedWatermark;
+        if (watermarkObjectStore.contains(UPDATED_WATERMARK_OS_KEY)) {
+          watermarkObjectStore.remove(UPDATED_WATERMARK_OS_KEY);
+        }
+        watermarkObjectStore.store(UPDATED_WATERMARK_OS_KEY, updatedWatermark);
+      } catch (ObjectStoreException e) {
+        throw new MuleRuntimeException(
+                                       createStaticMessage("An error occurred while trying to update the updatedWatermark in the the object store"),
```

Comment:
`"An error occurred while trying to update the updatedWatermark in the the object store with value: " + updatedWatermark`